java/util/Vector/Vector.hpp

```diff
@@ -0,0 +1,597 @@
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef NATIVE_JAVA_UTIL_VECTOR_HPP
+#define NATIVE_JAVA_UTIL_VECTOR_HPP
+
+#include "../../Lang.hpp"
+#include "../AbstractSequentialList/AbstractSequentialList.hpp"
+#include "../Collection/Collection.hpp"
+#include "../RandomAccess/RandomAccess.hpp"
+#include <initializer_list>
+
+using namespace Java::Lang;
+
+namespace Java {
+    namespace Util {
+        template <typename E>
+        class Vector
+//                :
+//                public virtual List<E>,
+//                public virtual RandomAccess,
+//                public virtual Cloneable,
+//                public virtual Serializable
+        {
+        protected:
+            std::vector<E> original;
+            int capacityIncrement;
+
+        public:
+            /**
+             * Vector Constructor
+             * Constructs an empty vector so that its internal data array has size 10 and its standard capacity increment is zero.
+             */
+            Vector() : Vector(10) { }
+
+            /**
+             * Vector Constructor
+             * Constructs an empty vector with the specified initial capacity and with its capacity increment equal to zero.
+             *
+             * @param initialCapacity
+             * @throw - if the specified initial capacity is negative
+             */
+            Vector(int initialCapacity) : Vector(initialCapacity, 0) { }
+
+            /**
+             * Vector Constructor
+             * Constructs an empty vector with the specified initial capacity and capacity increment.
+             *
+             * @param initialCapacity
+             * @param capacityIncrement
+             * @throw - if the specified initial capacity is negative
+             */
+            Vector(int initialCapacity, int capacityIncrement) {
+                if (initialCapacity < 0) {
+                    throw std::invalid_argument("initial capacity is negative");
+                }
+                this->original.reserve(initialCapacity);
+                this->capacityIncrement = capacityIncrement;
+            }
+
+            /**
+             * Vector Constructor
+             * Constructs a Vector containing the elements of the initializer list.
+             *
+             * @param list
+             */
+            Vector(const std::initializer_list<E> &list) {
```

Comment:
Please improve this method for performance
Should reserve one time with list size, not in loop